packages/flutter/test/material/elevated_button_test.dart

```diff
@@ -375,6 +375,82 @@ void main() {
     expect(didLongPressButton, isTrue);
   });
 
+   testWidgets('ElevatedButton.icon onPressed and onLongPress callbacks are correctly called when non-null', (WidgetTester tester) async {
+    bool wasPressed;
+    Finder elevatedButton;
+
+    Widget buildFrame({ VoidCallback? onPressed, VoidCallback? onLongPress }) {
+      return Directionality(
+        textDirection: TextDirection.ltr,
+        child: ElevatedButton.icon(
+          label: const Text('button'),
+          icon: const Icon(Icons.add),
+          onPressed: onPressed,
+          onLongPress: onLongPress,
+        ),
+      );
+    }
+
+    // onPressed not null, onLongPress null.
+    wasPressed = false;
+    await tester.pumpWidget(
+      buildFrame(onPressed: () { wasPressed = true; }, onLongPress: null),
+    );
+    elevatedButton = find.byType(ElevatedButton);
+    expect(tester.widget<ElevatedButton>(elevatedButton).enabled, true);
+    await tester.tap(elevatedButton);
+    expect(wasPressed, true);
+
+    // onPressed null, onLongPress not null.
+    wasPressed = false;
+    await tester.pumpWidget(
+      buildFrame(onPressed: null, onLongPress: () { wasPressed = true; }),
+    );
+    elevatedButton = find.byType(ElevatedButton);
+    expect(tester.widget<ElevatedButton>(elevatedButton).enabled, true);
+    await tester.longPress(elevatedButton);
+    expect(wasPressed, true);
+
+    // onPressed null, onLongPress null.
+    await tester.pumpWidget(
+      buildFrame(onPressed: null, onLongPress: null),
+    );
+    elevatedButton = find.byType(ElevatedButton);
+    expect(tester.widget<ElevatedButton>(elevatedButton).enabled, false);
+  });
+
+  testWidgets('ElevatedButton.icon onPressed and onLongPress callbacks are distinctly recognized', (WidgetTester tester) async {
```

Comment:
Again, this does not appear to test anything related to your change.

Reply:
ok @Piinks